File: app/src/main/java/ru/melodin/fast/api/model/VKVoice.java
package ru.melodin.fast.api.model;

import org.json.JSONArray;
import org.json.JSONObject;

import java.io.Serializable;
import java.util.ArrayList;

public class VKVoice extends VKModel implements Serializable {

    private static final long serialVersionUID = 1L;

    private int duration;
    private ArrayList<Integer> waveform;

    private String linkOgg, linkMp3;

    public VKVoice(JSONObject o) {
        duration = o.optInt("duration");
        linkMp3 = o.optString("link_mp3");
        linkOgg = o.optString("link_ogg");
        waveform = new ArrayList<>();

        JSONArray waveform = o.optJSONArray("waveform");
        for (int i = 0; i < waveform.length(); i++) {
            this.waveform.add((Integer) waveform.opt(i));
        }
    }

    public int getDuration() {
        return duration;
    }

    public ArrayList<Integer> getWaveform() {
        return waveform;
    }

    public String getLinkOgg() {
        return linkOgg;
    }

    public String getLinkMp3() {
        return linkMp3;
    }
}

File: app/src/main/java/ru/melodin/fast/api/model/VKVoice.kt
package ru.melodin.fast.api.model

import org.json.JSONObject
import java.io.Serializable
import java.util.*

class VKVoice(o: JSONObject) : VKModel(), Serializable {

    val duration: Int = o.optInt("duration")
    private val waveform: ArrayList<Int> = ArrayList()
    private val linkOgg: String = o.optString("link_ogg")
    val linkMp3: String = o.optString("link_mp3")

    init {

        val waveform = o.optJSONArray("waveform")
        for (i in 0 until waveform!!.length()) {
            this.waveform.add(waveform.opt(i) as Int)
        }
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
